### PR TITLE
Add GetMaxAttempts() to taskExecutionMetadata

### PR DESCRIPTION
**Pull request by catalinii**
https://github.com/lyft/flyte/issues/469

### CODE REVIEW

File: pkg/controller/nodes/task/taskexec_context.go [hunk 1]
```diff
@@ -140,13 +145,15 @@ func (t *Handler) newTaskExecutionContext(ctx context.Context, nCtx handler.Node
 	}
 
 	resourceNamespacePrefix := pluginCore.ResourceNamespace(t.resourceManager.GetID()).CreateSubNamespace(pluginCore.ResourceNamespace(pluginID))
+	maxAttempts := nCtx.Node().GetRetryStrategy().MinAttempts
```
Comment (anandswaminathan) [inline pkg/controller/nodes/task/taskexec_context.go:148]: @catalinii This field a bit tricky. Just see how it is used. The tricky part comes in interpretation of this field when you want to attempt 'x' times more or total 'x'

Comment (catalinii) [reply pkg/controller/nodes/task/taskexec_context.go]: It is total maxAttempts according to the code.

File: pkg/controller/nodes/task/taskexec_context.go [hunk 2]
```diff
@@ -140,13 +145,18 @@ func (t *Handler) newTaskExecutionContext(ctx context.Context, nCtx handler.Node
 	}
 
 	resourceNamespacePrefix := pluginCore.ResourceNamespace(t.resourceManager.GetID()).CreateSubNamespace(pluginCore.ResourceNamespace(pluginID))
+	maxAttempts := uint32(0)
```
Comment (anandswaminathan) [inline pkg/controller/nodes/task/taskexec_context.go:148]: I don't think this is valid. Attempt has to be greater than 0 right?

@katrogan @kumare3 Do we set default maxAttempts to 3 ?

Comment (katrogan) [reply pkg/controller/nodes/task/taskexec_context.go]: yeah it looks like we always set the min attempts to 1 + the user specified value: https://github.com/lyft/flytepropeller/blob/96c9ee9e2a9288101de846be8f49fce8c244ea2c/pkg/compiler/transformers/k8s/utils.go#L24

**Review by anandswaminathan — DISMISSED** [review]

File: pkg/controller/nodes/task/taskexec_context.go [hunk 3]
```diff
@@ -140,13 +145,18 @@ func (t *Handler) newTaskExecutionContext(ctx context.Context, nCtx handler.Node
 	}
 
 	resourceNamespacePrefix := pluginCore.ResourceNamespace(t.resourceManager.GetID()).CreateSubNamespace(pluginCore.ResourceNamespace(pluginID))
+	maxAttempts := uint32(3)
```
[the remark this answers is not in the document]
Comment (catalinii) [reply pkg/controller/nodes/task/taskexec_context.go]: Done

**Review by anandswaminathan — APPROVED** [review]

**Pull request merged**